internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1866,7 +1883,43 @@ func getStatusStateFromSpecState(state ramendrv1alpha1.ReplicationState) ramendr
 	}
 }
 
-func (v *VRGInstance) updateProtectedCGs() error {
+// nolint: dupl
+func (v *VRGInstance) updateProtectedCGsForVolSync(pvcGroups *[]ramendrv1alpha1.Groups) error {
+	var rgss ramendrv1alpha1.ReplicationGroupSourceList
+
+	matchLabels := map[string]string{
+		volsync.VRGOwnerNameLabel:      v.instance.Name,
+		volsync.VRGOwnerNamespaceLabel: v.instance.Namespace,
+	}
+
+	listOptions := []client.ListOption{
+		client.MatchingLabels(matchLabels),
+	}
+
+	if err := v.reconciler.List(v.ctx, &rgss, listOptions...); err != nil {
```

Comment:
Using the RGS list to find the groups works only if the ReplicationSources have already been created. For example, if the first sync never starts because of an issue with VGS, the groups won’t appear. A more reliable approach would be to use the same logic that originally creates the groups. That might be overkill though, and your current method could be good enough to address the issue. I just want to highlight that there’s a window where the groups won’t show up unless the first sync has started.

Reply:
agreed.